### PR TITLE
[plugins/aws][feat] Add Support for SNS

### LINKED ISSUE #1172: [AWS] Support SNS Service
### Motivation

Collect [AWS SNS](https://aws.amazon.com/sns) resources as part of the AWS collect cycle.

### Desired Behavior

Implement the resource collector as part of the [aws](https://github.com/someengineering/resoto/tree/main/plugins/aws) collector. To get an idea how a collector looks like, take the [SQS](https://github.com/someengineering/resoto/blob/main/plugins/aws/resoto_plugin_aws/resource/sqs.py) resource as inspiration. 

For further explanation and guidance, please visit the [Resource Guide](https://resoto.com/docs/edge/contributing/plugins/aws).

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by anjafr**
# Description

new resources:
- `aws_sns_topic`
- `aws_sns_subscription`
- `aws_sns_platform_application`
- `aws_sns_endpoint`

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Issues Fixed

- Fixes #1172 

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: plugins/aws/resoto_plugin_aws/resource/sns.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+from resoto_plugin_aws.resource.iam import AwsIamRole
+from resoto_plugin_aws.resource.kms import AwsKmsKey
+from resoto_plugin_aws.utils import ToDict
+from resotolib.baseresources import EdgeType, ModelReference
+from resotolib.json_bender import F, Bender, S, bend
+from resotolib.types import Json
+
+
+@define(eq=False, slots=False)
+class AwsSnsTopic(AwsResource):
```
Comment (aquamatthias) [inline plugins/aws/resoto_plugin_aws/resource/sns.py:14]: I wonder if we can introduce base classes here?

**Review by aquamatthias — APPROVED** [review]
LGTM - Thanks!

File: plugins/aws/resoto_plugin_aws/resource/sns.py (hunk 1 above)
Comment (anjafr) [reply plugins/aws/resoto_plugin_aws/resource/sns.py:14]: Interesting thought. Can't see right now what that would be and which other resources we might group together there? 🤔

Comment (aquamatthias) [reply plugins/aws/resoto_plugin_aws/resource/sns.py:14]: Probably worth a thought and discussion - maybe not as part of this PR.

**Pull request merged**